src/_pytest/config/__init__.py

```diff
@@ -999,24 +992,26 @@ class InvocationParams:
             Plugins accessing ``InvocationParams`` must be aware of that.
         """
 
-        args: Tuple[str, ...]
+        args: tuple[str, ...]
         """The command-line arguments as passed to :func:`pytest.main`."""
-        plugins: Optional[Sequence[Union[str, _PluggyPlugin]]]
+        plugins: Sequence[str | _PluggyPlugin] | None
         """Extra plugins, might be `None`."""
-        dir: Path
-        """The directory from which :func:`pytest.main` was invoked."""
+        dir: pathlib.Path
+        """The directory from which :func:`pytest.main` was invoked. :type: pathlib.Path"""
 
         def __init__(
             self,
             *,
             args: Iterable[str],
-            plugins: Optional[Sequence[Union[str, _PluggyPlugin]]],
-            dir: Path,
+            plugins: Sequence[str | _PluggyPlugin] | None,
+            dir: pathlib.Path,
         ) -> None:
             object.__setattr__(self, "args", tuple(args))
             object.__setattr__(self, "plugins", plugins)
             object.__setattr__(self, "dir", dir)
 
+    cache: Cache
```

Comment:
```suggestion
    # Set by cacheprovider plugin.
    cache: Cache
```

for consistency with `main` (and less confusion).

Also, `main` has `Cache | None` here, but I *think* this is actually more correct? Looking at [`pytest_configure` in `cacheprovider.py`](https://github.com/pytest-dev/pytest/blob/f426c0b35a8bd3af253799159ecd9c78bb5ea1a2/src/_pytest/cacheprovider.py#L554) I don't see a reason why this should ever be `None`...